test/gomod.collector.test.ts

```diff
@@ -195,28 +195,84 @@ github.com/btcsuite/btcd`);
 
   it('tests replace statements in go.mod', async () => {
```

Comment:
Can we add below test cases
1. Pseudo version to semver replacement.
2. Semver to pseudo version replacement.
3.  Single line replace statement in go.mod.
4. Multiple single line replace statements.
5. Replace block placed before require section.
6. Few replace statement before require and after required section.
7. Replace with local path in right side like absolute and relative path for replaced module. 
8. Some replace statement which are commented, those should not be considered.